### PR TITLE
Replace FileChooserButton for GTK4

### PR DESCRIPTION
**Pull request by treppenwitz03**
This hopes to check a todo in https://github.com/elementary/screenshot/pull/240 so screenshot can be ported.

### CODE REVIEW

**Comment by jeremypw** [comment]
Maybe should stick to the FileChooserNative API as this is now Flatpak'd?

**Comment by jeremypw** [comment]
Screenshot from 2022-05-31 18 03 40

The drop arrows on the two buttons are of a different size, furthermore the filechooser button does not produce a dropdown when clicked, but a dialog so I think it should have an ellipsis?

File: src/Widgets/SaveDialog.vala [hunk 1]
```diff
@@ -131,8 +131,45 @@ public class Screenshot.SaveDialog : Granite.Dialog {
...
+        var location_button = new Gtk.Button ();
+        location_button.add (location_button_indicator);
+
+        var location_dialog = new Gtk.FileChooserNative (_("Select Screenshots Folder…"), this,
+            Gtk.FileChooserAction.SELECT_FOLDER, "Open", "Cancel");
+        location_dialog.set_current_folder (folder_dir);
+
+        var size_group = new Gtk.SizeGroup (Gtk.SizeGroupMode.VERTICAL);
+        size_group.add_widget (format_cmb);
+        size_group.add_widget (location_button);
```
Comment (jeremypw) [inline src/Widgets/SaveDialog.vala:173]: You can avoid using a size group by taking the `preview_box` out of the grid and then applying `row_homogeneous = true` to the grid (and not making the top margin zero).

**Comment by jeremypw** [comment]
Screenshot from 2022-05-31 18 44 31

This is how it looks after applying my suggestions.

**Review by jeremypw — CHANGES_REQUESTED** [review]

File: src/Widgets/SaveDialog.vala [hunk 2]
```diff
@@ -131,8 +131,45 @@ public class Screenshot.SaveDialog : Granite.Dialog {
...
+        };
+
+        if (folder_dir == Environment.get_home_dir ()) {
+            folder_name.label = "Home";
+            folder_icon.gicon = new ThemedIcon ("user-home");
+        } else if (folder_dir == "/") {
+            folder_name.label = "File System";
+            folder_icon.gicon = new ThemedIcon ("drive-harddisk");
+        } else {
+            folder_name.label = folder_dir.reverse ().split ("/")[0].reverse ();
```
[the remark this answers is not in the document]
Comment (ZenitsuDev) [reply src/Widgets/SaveDialog.vala:149]: That looks better. Thanks

File: src/Widgets/SaveDialog.vala [hunk 3]
```diff
@@ -131,8 +131,45 @@ public class Screenshot.SaveDialog : Granite.Dialog {
...
+
+        var arrow = new Gtk.Image () {
+            gicon = new ThemedIcon ("pan-down-symbolic"),
+            halign = Gtk.Align.END
+        };
+
+        var location_button_indicator = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 0);
+        location_button_indicator.pack_start (folder_icon, false, false);
+        location_button_indicator.pack_start (folder_name, true, true);
+        location_button_indicator.pack_start (arrow, false, false);
```
[the remark this answers is not in the document]
Comment (ZenitsuDev) [reply src/Widgets/SaveDialog.vala:161]: Understood. I will change it thanks

File: src/Widgets/SaveDialog.vala (hunk 1 above)
Comment (ZenitsuDev) [reply src/Widgets/SaveDialog.vala:173]: Great Idea. Thanks

**Comment by ZenitsuDev** [comment]
oh I found it! it is `content-loading-symbolic`

**Comment by ZenitsuDev** [comment]
Screenshot from 2022-06-01 04 30 48

it does not seem aligned still.

**Comment by jeremypw** [comment]
Probably best seek @danrabbit 's opinion on the finer points of the design.

**Comment by jeremypw** [comment]
Still being held up by the existence of #253.

**Review by danirabbit — APPROVED** [review]
Works as expected, nice job!

**Pull request merged**